capsules/src/hmac.rs

```diff
@@ -0,0 +1,503 @@
+                                        for (i, d) in d
+                                            .as_ref()
+                                            .get(copied_data..)
+                                            .unwrap()
+                                            .iter()
+                                            .enumerate()
+                                        {
+                                            buf[i] = *d;
+                                        }
+                                    } else {
+                                        // Copy the entire buffers worth of data
+                                        buf.copy_from_slice(
+                                            d.as_ref()
+                                                .get(copied_data..(copied_data + static_buffer))
+                                                .unwrap(),
+                                        );
+                                    }
+                                }
+                            }
+                            None => {
+                                return;
+                            }
+                        }
+                    });
+
+                    if static_buffer > 0 {
+                        let copied_data = self.data_copied.get();
+
+                        if data_len > copied_data {
+                            // Update the amount of data copied
+                            self.data_copied.set(copied_data + static_buffer);
+
+                            // Add the data from the static buffer to the HMAC
+                            if data_len < (copied_data + static_buffer) {
+                                if let Err(_e) = self.hmac.add_data(
+                                    self.data_buffer
+                                        .take()
+                                        .unwrap()
+                                        .get_mut(0..(data_len - copied_data))
+                                        .unwrap(),
+                                ) {
+                                    return;
+                                }
+                            } else {
+                                if let Err(_e) =
+                                    self.hmac.add_data(self.data_buffer.take().unwrap())
+                                {
+                                    return;
+                                }
+                            }
+
+                            // Return as we don't want to run the digest yet
+                            return;
+                        }
+                    }
+
+                    // If we get here we are ready to run the digest, reset the copied data
+                    self.data_copied.set(0);
+
+                    if let Err(_e) = self.hmac.run(self.dest_buffer.take().unwrap()) {
+                        app.callback.map(|cb| {
+                            // cb.schedule(e.0 as usize, e.1, 0);
```

Comment:
What is this commented line for?

Reply:
I have fixed this, I was meant to call the commented function.